jax/lax_linalg.py

```diff
@@ -382,13 +382,28 @@ def triangular_solve_batching_rule(batched_args, batch_dims, left_side,
                                    unit_diagonal):
   x, y = batched_args
   bx, by = batch_dims
-  size = next(t.shape[i] for t, i in zip(batched_args, batch_dims)
-              if i is not None)
-  x = batching.bdim_at_front(x, bx, size)
-  y = batching.bdim_at_front(y, by, size)
-  return triangular_solve(x, y, left_side=left_side, lower=lower,
-                          transpose_a=transpose_a, conjugate_a=conjugate_a,
-                          unit_diagonal=unit_diagonal), 0
+  if bx is batching.not_mapped:
+    if left_side:
+      y = batching.moveaxis(y, by, -1)
```

Comment:
This isn't strictly related to the PR, but reading through this I couldn't remember whether the primitive allows `b` to be a vector (it seems it does not). But reading the shape rule, it does not actually check directly that `b` has at least 2 dimensions. It might help to add that check as both documentation and context for this kind of rule, and because that's an actual bug in the shape rule.

Reply:
Added an explicit check (without a test, but, you know, that's still better than nothing!)